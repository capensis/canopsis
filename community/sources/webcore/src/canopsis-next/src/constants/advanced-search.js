export const ADVANCED_SEARCH_ITEM_TYPES = {
  field: 'field',
  condition: 'condition',
  value: 'value',
  union: 'union',
};

export const ADVANCED_SEARCH_NEXT_ITEM_TYPES = {
  [ADVANCED_SEARCH_ITEM_TYPES.field]: ADVANCED_SEARCH_ITEM_TYPES.condition,
  [ADVANCED_SEARCH_ITEM_TYPES.condition]: ADVANCED_SEARCH_ITEM_TYPES.value,
  [ADVANCED_SEARCH_ITEM_TYPES.value]: ADVANCED_SEARCH_ITEM_TYPES.union,
  [ADVANCED_SEARCH_ITEM_TYPES.union]: ADVANCED_SEARCH_ITEM_TYPES.field,
};

export const ADVANCED_SEARCH_NOT = 'NOT';

export const ADVANCED_SEARCH_UNION_CONDITIONS = {
  and: 'AND',
  or: 'OR',
};

export const ADVANCED_SEARCH_CONDITIONS = {
  less: '<',
  more: '>',
  equal: '=',
  notEqual: '!=',
  like: 'LIKE',
  contains: 'CONTAINS',
};

export const ADVANCED_SEARCH_UNION_REGEXP_PATTERN = new RegExp(`\\s(${Object.values(ADVANCED_SEARCH_UNION_CONDITIONS).join('|')})(\\s|$)`, 'gi');

export const ADVANCED_SEARCH_UNION_FIELDS = [
  {
    value: ADVANCED_SEARCH_UNION_CONDITIONS.and,
    type: ADVANCED_SEARCH_ITEM_TYPES.union,
    text: ADVANCED_SEARCH_UNION_CONDITIONS.and,
  },
  {
    value: ADVANCED_SEARCH_UNION_CONDITIONS.or,
    type: ADVANCED_SEARCH_ITEM_TYPES.union,
    text: ADVANCED_SEARCH_UNION_CONDITIONS.or,
  },
];
