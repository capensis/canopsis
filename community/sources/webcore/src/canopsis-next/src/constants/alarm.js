import { COLORS } from '@/config';

export const ALARM_FIELDS = {
  id: '_id',
  displayName: 'v.display_name',
  output: 'v.output',
  longOutput: 'v.long_output',
  initialOutput: 'v.initial_output',
  initialLongOutput: 'v.initial_long_output',
  connector: 'v.connector',
  connectorName: 'v.connector_name',
  component: 'v.component',
  resource: 'v.resource',
  lastComment: 'v.last_comment.m',
  ackBy: 'v.ack.a',
  ackMessage: 'v.ack.m',
  ackInitiator: 'v.ack.initiator',
  stateMessage: 'v.state.m',
  statusMessage: 'v.status.m',
  state: 'v.state.val',
  status: 'v.status.val',
  totalStateChanges: 'v.total_state_changes',
  timestamp: 't',
  creationDate: 'v.creation_date',
  lastUpdateDate: 'v.last_update_date',
  lastEventDate: 'v.last_event_date',
  ackAt: 'v.ack.t',
  stateAt: 'v.state.t',
  statusAt: 'v.status.t',
  resolved: 'v.resolved',
  activationDate: 'v.activation_date',
  duration: 'v.duration',
  currentStateDuration: 'v.current_state_duration',
  snoozeDuration: 'v.snooze_duration',
  pbhInactiveDuration: 'v.pbh_inactive_duration',
  activeDuration: 'v.active_duration',
  eventsCount: 'v.events_count',
  tags: 'tags',
  extraDetails: 'extra_details',
  impactState: 'impact_state',
  infos: 'v.infos',
  links: 'links',
  entityId: 'entity._id',
  entityName: 'entity.name',
  entityCategoryName: 'entity.category.name',
  entityType: 'entity.type',
  entityComponent: 'entity.component',
  entityConnector: 'entity.connector',
  entityImpactLevel: 'entity.impact_level',
  entityKoEvents: 'entity.ko_events',
  entityOkEvents: 'entity.ok_events',
  entityInfos: 'entity.infos',
  entityComponentInfos: 'entity.component_infos',
  entityLastPbehaviorDate: 'entity.last_pbehavior_date',

  /**
   * OBJECTS
   */
  ack: 'v.ack',
  ticket: 'v.ticket',
  canceled: 'v.canceled',
  snooze: 'v.snooze',
  pbehaviorInfo: 'v.pbehavior_info',

  /**
   * VIRTUAL
   */
  activated: 'activated',
};

export const ALARM_INFOS_FIELDS = [
  ALARM_FIELDS.infos,
  ALARM_FIELDS.entityInfos,
  ALARM_FIELDS.entityComponentInfos,
];

export const ALARM_LEVELS = {
  minor: 20,
  major: 30,
  critical: 40,
};

export const ALARM_LEVELS_COLORS = {
  ok: COLORS.state.ok,
  minor: COLORS.state.minor,
  major: COLORS.state.major,
  critical: COLORS.state.critical,
};

export const ALARM_LIST_ACTIONS_TYPES = {
  ack: 'ack',
  fastAck: 'fastAck',
  ackRemove: 'ackRemove',
  pbehaviorAdd: 'pbehaviorAdd',
  moreInfos: 'moreInfos',
  snooze: 'snooze',
  declareTicket: 'declareTicket',
  associateTicket: 'associateTicket',
  cancel: 'cancel',
  changeState: 'changeState',
  variablesHelp: 'variablesHelp',
  history: 'history',
  groupRequest: 'groupRequest',
  createManualMetaAlarm: 'createManualMetaAlarm',
  removeAlarmsFromManualMetaAlarm: 'removeAlarmsFromManualMetaAlarm',
  updateManualMetaAlarm: 'updateManualMetaAlarm',
  comment: 'comment',

  links: 'links',

  correlation: 'correlation',

  listFilters: 'listFilters',
  editFilter: 'editFilter',
  addFilter: 'addFilter',
  userFilter: 'userFilter',

  listRemediationInstructionsFilters: 'listRemediationInstructionsFilters',
  editRemediationInstructionsFilter: 'editRemediationInstructionsFilter',
  addRemediationInstructionsFilter: 'addRemediationInstructionsFilter',
  userRemediationInstructionsFilter: 'userRemediationInstructionsFilter',

  executeInstruction: 'executeInstruction',
};

export const META_ALARMS_RULE_TYPES = {
  relation: 'relation',
  timebased: 'timebased',
  attribute: 'attribute',
  complex: 'complex',
  valuegroup: 'valuegroup',
  corel: 'corel',

  /**
   * Manual group type doesn't use in the form
   * We are using it only inside alarms list widget
   */
  manualgroup: 'manualgroup',
};

export const META_ALARMS_THRESHOLD_TYPES = {
  thresholdRate: 'thresholdRate',
  thresholdCount: 'thresholdCount',
};

export const ALARMS_LIST_TIME_LINE_SYSTEM_AUTHOR = 'canopsis.engine';

export const DEFAULT_ALARMS_WIDGET_COLUMNS = [
  { value: ALARM_FIELDS.connector },
  { value: ALARM_FIELDS.connectorName },
  { value: ALARM_FIELDS.component },
  { value: ALARM_FIELDS.resource },
  { value: ALARM_FIELDS.output },
  { value: ALARM_FIELDS.extraDetails },
  { value: ALARM_FIELDS.state },
  { value: ALARM_FIELDS.status },
];

export const DEFAULT_ALARMS_WIDGET_GROUP_COLUMNS = [
  { value: ALARM_FIELDS.connector },
  { value: ALARM_FIELDS.connectorName },
  { value: ALARM_FIELDS.resource },
  { value: ALARM_FIELDS.output },
  { value: ALARM_FIELDS.extraDetails },
  { value: ALARM_FIELDS.state },
  { value: ALARM_FIELDS.status },
];

export const DEFAULT_CONTEXT_WIDGET_RESOLVED_ALARMS_COLUMNS = [
  { value: ALARM_FIELDS.id },
  { value: ALARM_FIELDS.creationDate },
  { value: ALARM_FIELDS.lastUpdateDate },
];

export const DEFAULT_CONTEXT_WIDGET_ACTIVE_ALARM_COLUMNS = [
  { value: ALARM_FIELDS.displayName },
  { value: ALARM_FIELDS.output },
  { value: ALARM_FIELDS.state },
  { value: ALARM_FIELDS.status },
];

export const ALARMS_OPENED_VALUES = {
  opened: true,
  all: null,
  resolved: false,
};

export const ALARM_METRIC_PARAMETERS = {
  activeAlarms: 'active_alarms',
  createdAlarms: 'created_alarms',
  nonDisplayedAlarms: 'non_displayed_alarms',
  instructionAlarms: 'instruction_alarms',
  pbehaviorAlarms: 'pbehavior_alarms',
  correlationAlarms: 'correlation_alarms',
  ackAlarms: 'ack_alarms',
  cancelAckAlarms: 'cancel_ack_alarms',
  ackActiveAlarms: 'ack_active_alarms',
  ticketActiveAlarms: 'ticket_active_alarms',
  withoutTicketActiveAlarms: 'without_ticket_active_alarms',
  ratioCorrelation: 'ratio_correlation',
  ratioInstructions: 'ratio_instructions',
  ratioTickets: 'ratio_tickets',
  ratioNonDisplayed: 'ratio_non_displayed',
  ratioRemediatedAlarms: 'ratio_remediated_alarms',
  averageAck: 'average_ack',
  averageResolve: 'average_resolve',
  manualInstructionAssignedAlarms: 'manual_instruction_assigned_alarms',
  manualInstructionExecutedAlarms: 'manual_instruction_executed_alarms',
};

export const ALARMS_LIST_HEADER_OPACITY_DELAY = 500;

export const ALARM_INTERVAL_FIELDS = {
  timestamp: 't',
  resolved: 'v.resolved',
  lastUpdateDate: 'v.last_update_date',
  lastEventDate: 'v.last_event_date',
};

const {
  ack,
  ticket,
  canceled,
  snooze,
  pbehaviorInfo,
  activated,

  ...alarmListWidgetColumns
} = ALARM_FIELDS;

export const ALARM_LIST_WIDGET_COLUMNS = alarmListWidgetColumns;

export const ALARM_PATTERN_FIELDS = {
  displayName: ALARM_FIELDS.displayName,
  state: ALARM_FIELDS.state,
  status: ALARM_FIELDS.status,
  component: ALARM_FIELDS.component,
  resource: ALARM_FIELDS.resource,
  connector: ALARM_FIELDS.connector,
  connectorName: ALARM_FIELDS.connectorName,
  creationDate: ALARM_FIELDS.creationDate,
  duration: ALARM_FIELDS.duration,
  infos: ALARM_FIELDS.infos,
  output: ALARM_FIELDS.output,
  lastEventDate: ALARM_FIELDS.lastEventDate,
  lastUpdateDate: ALARM_FIELDS.lastUpdateDate,
  activationDate: ALARM_FIELDS.activationDate,
  ack: ALARM_FIELDS.ack,
  ackAt: ALARM_FIELDS.ackAt,
  ackBy: ALARM_FIELDS.ackBy,
  ackMessage: ALARM_FIELDS.ackMessage,
  ackInitiator: ALARM_FIELDS.ackInitiator,
  resolved: ALARM_FIELDS.resolved,
  ticket: ALARM_FIELDS.ticket,
  canceled: ALARM_FIELDS.canceled,
  snooze: ALARM_FIELDS.snooze,
  lastComment: ALARM_FIELDS.lastComment,
  longOutput: ALARM_FIELDS.longOutput,
  initialOutput: ALARM_FIELDS.initialOutput,
  initialLongOutput: ALARM_FIELDS.initialLongOutput,
  totalStateChanges: ALARM_FIELDS.totalStateChanges,
  tags: ALARM_FIELDS.tags,
  activated: ALARM_FIELDS.activated,
};

export const ALARM_ACK_INITIATORS = {
  user: 'user',
  system: 'system',
  external: 'external',
};

export const ALARM_STEP_FIELDS = {
  timestamp: 't',
  value: 'val',
  message: 'm',
  author: 'a',
};

export const ALARM_TEMPLATE_FIELDS = {
  id: `alarm.${ALARM_FIELDS.id}`,
  ack: `alarm.${ALARM_FIELDS.ack}`,
  state: `alarm.${ALARM_FIELDS.state}`,
  status: `alarm.${ALARM_FIELDS.status}`,
  ticket: `alarm.${ALARM_FIELDS.ticket}`,
  component: `alarm.${ALARM_FIELDS.component}`,
  connector: `alarm.${ALARM_FIELDS.connector}`,
  connectorName: `alarm.${ALARM_FIELDS.connectorName}`,
  resource: `alarm.${ALARM_FIELDS.resource}`,
  creationDate: `alarm.${ALARM_FIELDS.creationDate}`,
  displayName: `alarm.${ALARM_FIELDS.displayName}`,
  output: `alarm.${ALARM_FIELDS.output}`,
  lastUpdateDate: `alarm.${ALARM_FIELDS.lastUpdateDate}`,
  lastEventDate: `alarm.${ALARM_FIELDS.lastEventDate}`,
  pbehaviorInfo: `alarm.${ALARM_FIELDS.pbehaviorInfo}`,
  duration: `alarm.${ALARM_FIELDS.duration}`,
  eventsCount: `alarm.${ALARM_FIELDS.eventsCount}`,
};

export const ALARM_FIELDS_TO_LABELS_KEYS = {
  [ALARM_FIELDS.id]: 'common.id',
  [ALARM_FIELDS.displayName]: 'alarm.fields.displayName',
  [ALARM_FIELDS.output]: 'common.output',
  [ALARM_FIELDS.longOutput]: 'common.longOutput',
  [ALARM_FIELDS.initialOutput]: 'alarm.fields.initialOutput',
  [ALARM_FIELDS.initialLongOutput]: 'alarm.fields.initialLongOutput',
  [ALARM_FIELDS.connector]: 'common.connector',
  [ALARM_FIELDS.connectorName]: 'common.connectorName',
  [ALARM_FIELDS.component]: 'common.component',
  [ALARM_FIELDS.resource]: 'common.resource',
  [ALARM_FIELDS.lastComment]: 'alarm.fields.lastComment',
  [ALARM_FIELDS.ackBy]: 'alarm.fields.ackBy',
  [ALARM_FIELDS.ackMessage]: 'alarm.fields.ackMessage',
  [ALARM_FIELDS.ackInitiator]: 'alarm.fields.ackInitiator',
  [ALARM_FIELDS.stateMessage]: 'alarm.fields.stateMessage',
  [ALARM_FIELDS.statusMessage]: 'alarm.fields.statusMessage',
  [ALARM_FIELDS.state]: 'common.state',
  [ALARM_FIELDS.status]: 'common.status',
  [ALARM_FIELDS.totalStateChanges]: 'alarm.fields.totalStateChanges',
  [ALARM_FIELDS.timestamp]: 'common.timestamp',
  [ALARM_FIELDS.creationDate]: 'common.created',
  [ALARM_FIELDS.lastUpdateDate]: 'common.updated',
  [ALARM_FIELDS.lastEventDate]: 'common.lastEventDate',
  [ALARM_FIELDS.ackAt]: 'alarm.fields.ackAt',
  [ALARM_FIELDS.stateAt]: 'alarm.fields.stateAt',
  [ALARM_FIELDS.statusAt]: 'alarm.fields.statusAt',
  [ALARM_FIELDS.resolved]: 'alarm.fields.resolved',
  [ALARM_FIELDS.activationDate]: 'alarm.fields.activationDate',
  [ALARM_FIELDS.duration]: 'common.duration',
  [ALARM_FIELDS.currentStateDuration]: 'alarm.fields.currentStateDuration',
  [ALARM_FIELDS.snoozeDuration]: 'alarm.fields.snoozeDuration',
  [ALARM_FIELDS.pbhInactiveDuration]: 'alarm.fields.pbhInactiveDuration',
  [ALARM_FIELDS.activeDuration]: 'alarm.fields.activeDuration',
  [ALARM_FIELDS.eventsCount]: 'alarm.fields.eventsCount',
  [ALARM_FIELDS.tags]: 'common.tag',
  [ALARM_FIELDS.extraDetails]: 'alarm.fields.extraDetails',
  [ALARM_FIELDS.impactState]: 'common.impactState',
  [ALARM_FIELDS.infos]: 'common.infos',
  [ALARM_FIELDS.links]: 'common.link',
  [ALARM_FIELDS.entityId]: 'alarm.fields.entityId',
  [ALARM_FIELDS.entityName]: 'alarm.fields.entityName',
  [ALARM_FIELDS.entityCategoryName]: 'alarm.fields.entityCategoryName',
  [ALARM_FIELDS.entityType]: 'alarm.fields.entityType',
  [ALARM_FIELDS.entityComponent]: 'alarm.fields.entityComponent',
  [ALARM_FIELDS.entityConnector]: 'alarm.fields.entityConnector',
  [ALARM_FIELDS.entityImpactLevel]: 'alarm.fields.entityImpactLevel',
  [ALARM_FIELDS.entityKoEvents]: 'alarm.fields.entityKoEvents',
  [ALARM_FIELDS.entityOkEvents]: 'alarm.fields.entityOkEvents',
  [ALARM_FIELDS.entityInfos]: 'alarm.fields.entityInfos',
  [ALARM_FIELDS.entityComponentInfos]: 'alarm.fields.entityComponentInfos',
  [ALARM_FIELDS.entityLastPbehaviorDate]: 'alarm.fields.entityLastPbehaviorDate',

  /**
   * OBJECTS
   */
  [ALARM_FIELDS.ack]: 'common.ack',
  [ALARM_FIELDS.ticket]: 'common.ticket',
  [ALARM_FIELDS.canceled]: 'common.canceled',
  [ALARM_FIELDS.snooze]: 'common.snooze',
  [ALARM_FIELDS.pbehaviorInfo]: 'pbehavior.pbehaviorInfo',

  /**
   * VIRTUAL
   */
  [ALARM_FIELDS.activated]: 'common.activated',
};

export const ALARM_UNSORTABLE_FIELDS = [
  ALARM_FIELDS.extraDetails,
  ALARM_FIELDS.links,
  ALARM_FIELDS.tags,
];

export const ALARM_DENSE_TYPES = {
  large: 0,
  medium: 1,
  small: 2,
};

export const ALARM_PAYLOADS_VARIABLES = {
  alarm: '.Alarm',
  alarms: '.Alarms',
  component: '.Value.Component',
  resource: '.Value.Resource',
  stateMessage: '.Value.State.Message',
  stateValue: '.Value.State.Value',
  statusValue: '.Value.Status.Value',
  ticketAuthor: '.Value.Ticket.Author',
  ticketValue: '.Value.Ticket.Ticket',
  ticketMessage: '.Value.Ticket.Message',
  ackAuthor: '.Value.ACK.Author',
  ackMessage: '.Value.ACK.Message',
  lastCommentAuthor: '.Value.LastComment.Author',
  lastCommentMessage: '.Value.LastComment.Message',

  entityName: '.Entity.Name',
  entityInfosValue: '(index .Entity.Infos.%infos_name%).Value',
};

export const ACK_MODAL_ACTIONS_TYPES = {
  ack: 0,
  ackAndAssociateTicket: 1,
  ackAndDeclareTicket: 2,
};

export const ALARMS_EXPAND_PANEL_TABS = {
  moreInfos: 'moreInfos',
  timeLine: 'timeLine',
  ticketsDeclared: 'ticketsDeclared',
  pbehavior: 'pbehavior',
  alarmsChildren: 'alarmsChildren',
  trackSource: 'trackSource',
  impactChain: 'impactChain',
  entityGantt: 'entityGantt',
};
