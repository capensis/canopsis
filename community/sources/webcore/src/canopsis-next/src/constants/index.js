export * from './recurrence-rule';
export * from './common';
export * from './entity';
export * from './alarm';
export * from './broadcast-message';
export * from './date';
export * from './dynamic-info';
export * from './event-filter';
export * from './pattern';
export * from './healthcheck';
export * from './idle-rule';
export * from './junit';
export * from './modal';
export * from './pbehavior';
export * from './permission';
export * from './remediation';
export * from './scenario';
export * from './snmp';
export * from './stats';
export * from './store';
export * from './widget';
export * from './kpi';
export * from './user';
export * from './service-weather';
export * from './map';
export * from './flowchart';
export * from './parameters';
export * from './declare-ticket';
export * from './link-rule';
export * from './external-data';
export * from './tag';
export * from './counter';
export * from './calendar';
export * from './icon';
export * from './availability';
