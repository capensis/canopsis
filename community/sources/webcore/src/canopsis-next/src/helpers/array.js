import { omit, pick, isObject, map } from 'lodash';

import { uid } from './uid';

/**
 * Add uniq key field in entity object.
 *
 * @param {Object} entity
 * @return {Object}
 */
export const addKeyInEntity = (entity = {}) => ({ ...entity, key: uid() });

/**
 * Add uniq key field in each entity.
 *
 * @param {Array} entities
 * @return {Array}
 */
export const addKeyInEntities = (entities = []) => entities.map(addKeyInEntity);

/**
 * Remove key field from each entity.
 *
 * @param {Object} entity
 * @return {Object}
 */
export const removeKeyFromEntity = (entity = {}) => omit(entity, ['key']);

/**
 * Remove key field from each entity.
 *
 * @param {Array} entities
 * @return {Array}
 */
export const removeKeyFromEntities = (entities = []) => entities.map(removeKeyFromEntity);

/**
 * Get id from entity
 *
 * @param {Object} entity
 * @param {string} idField
 * @return {string}
 */
export const getIdFromEntity = (entity, idField = '_id') => (isObject(entity) ? entity[idField] : entity);

/**
 * Return entities ids
 *
 * @param {Array} entities
 * @param {string} [idKey = '_id']
 */
export const mapIds = (entities, idKey = '_id') => map(entities, idKey);

/**
 * Pick id field from entities
 *
 * @param {U[]} entities
 * @param {string} idKey
 * @returns {PartialObject<U>[]}
 */
export const pickIds = (entities = [], idKey = '_id') => entities.map(entity => pick(entity, [idKey]));

/**
 * Filter entities by ids
 *
 * @param {Object[]} items
 * @param {Object} item
 * @param {string} [idKey = '_id']
 */
export const filterById = (items, item, idKey = '_id') => items
  .filter(({ [idKey]: itemId }) => item[idKey] !== itemId);

/**
 * Filter entities by value
 *
 * @param {string[] | number[]} items
 * @param {string | number} removingValue
 */
export const filterValue = (items, removingValue) => items.filter(item => item !== removingValue);

/**
 * Create range number array
 *
 * @param {number} length
 * @param {number} [start = 0]
 * @return {number[]}
 */
export const createRangeArray = (length, start = 0) => Array.from({ length }, (_, index) => start + index);
