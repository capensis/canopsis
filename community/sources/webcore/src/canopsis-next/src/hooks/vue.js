import { getCurrentInstance } from 'vue';

/**
 * Hook for get current component instance
 *
 * @return {Vue}
 */
export const useComponentInstance = () => {
  const instance = getCurrentInstance();

  return instance?.proxy || instance;
};

/**
 * Hook for get current component options
 *
 * @return {Object}
 */
export const useComponentOptions = () => {
  const instance = useComponentInstance();

  return instance.$options;
};

/**
 * Hook for get current component model
 *
 * @return {Object}
 */
export const useComponentModel = () => {
  const { model } = useComponentOptions();

  return {
    event: model?.event ?? 'input',
    prop: model?.event ?? 'value',
  };
};
