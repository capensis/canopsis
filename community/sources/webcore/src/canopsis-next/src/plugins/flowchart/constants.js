export const DIRECTIONS = {
  northWest: 'nw',
  north: 'n',
  northEast: 'ne',
  east: 'e',
  southEast: 'se',
  south: 's',
  southWest: 'sw',
  west: 'w',
};
