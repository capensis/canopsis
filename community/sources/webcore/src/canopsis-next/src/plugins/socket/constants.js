export const REQUEST_MESSAGES_TYPES = {
  ping: 0,
  join: 1,
  leave: 2,
  authenticate: 3,
};

export const RESPONSE_MESSAGES_TYPES = {
  pong: 0,
  ok: 1,
  error: 2,
  close: 3,
  authenticated: 4,
};

export const ICONS_RESPONSE_MESSAGES_TYPES = {
  create: 0,
  update: 1,
  delete: 2,
};

export const MAX_RECONNECTS_COUNT = 10;

export const PING_INTERVAL = 5000;

export const RECONNECT_INTERVAL = 5000;

export const EVENTS_TYPES = {
  customClose: 'custom-close',
  closeRoom: 'close-room',
  networkError: 'network-error',
};

export const ERROR_MESSAGES = {
  authenticationFailed: 'authentication failed',
};
