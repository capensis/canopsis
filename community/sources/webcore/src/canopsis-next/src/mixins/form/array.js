import { setField } from '@/helpers/immutable';

import { formBaseMixin, modelPropKeyComputed } from './base';

/**
 * @mixin Form mixin
 * @deprecated Should be used useArrayModelField
 */
export const formArrayMixin = {
  mixins: [formBaseMixin],
  methods: {
    /**
     * Emit event to parent with new array with new item
     *
     * @param {*} value
     * @return {Array}
     */
    addItemIntoArray(value) {
      return this.updateModel([...this[this[modelPropKeyComputed]], value]);
    },

    /**
     * Emit event to parent with new array with updated array item
     *
     * @param {number} index
     * @param {*} value
     * @return {Array}
     */
    updateItemInArray(index, value) {
      const items = [...this[this[modelPropKeyComputed]]];

      items[index] = value;

      return this.updateModel(items);
    },

    /**
     * Emit event to parent with new array with updated array item with updated field
     *
     * @param {number} index
     * @param {string} fieldName
     * @param {*} value
     * @return {Array}
     */
    updateFieldInArrayItem(index, fieldName, value) {
      return this.updateItemInArray(index, setField(this[this[modelPropKeyComputed]][index], fieldName, value));
    },

    /**
     * Emit event to parent with new array without array item
     *
     * @param {number} index
     * @return {Array}
     */
    removeItemFromArray(index) {
      return this.updateModel(this[this[modelPropKeyComputed]].filter((v, i) => i !== index));
    },
  },
};

export default formArrayMixin;
