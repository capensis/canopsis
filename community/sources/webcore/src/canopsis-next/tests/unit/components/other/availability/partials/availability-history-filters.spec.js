import { generateShallowRenderer, generateRenderer } from '@unit/utils/vue';
import { randomArrayItem } from '@unit/utils/array';

import { AVAILABILITY_SHOW_TYPE, QUICK_RANGES, SAMPLINGS } from '@/constants';

import AvailabilityHistoryFilters from '@/components/other/availability/partials/availability-history-filters.vue';

const stubs = {
  'c-sampling-field': true,
  'availability-show-type-field': true,
};

const selectSamplingField = wrapper => wrapper.find('c-sampling-field-stub');
const selectAvailabilityShowTypeField = wrapper => wrapper.find('availability-show-type-field-stub');

describe('availability-history-filters', () => {
  const factory = generateShallowRenderer(AvailabilityHistoryFilters, { stubs });
  const snapshotFactory = generateRenderer(AvailabilityHistoryFilters, { stubs });

  test('Sampling changed after trigger sampling field', async () => {
    const wrapper = factory();

    const newValue = randomArrayItem(Object.values(SAMPLINGS));

    await selectSamplingField(wrapper).triggerCustomEvent('input', newValue);

    expect(wrapper).toEmit('update:sampling', newValue);
  });

  test('Show type changed after trigger show type field', async () => {
    const wrapper = factory();

    const newValue = randomArrayItem(Object.values(AVAILABILITY_SHOW_TYPE));

    await selectAvailabilityShowTypeField(wrapper).triggerCustomEvent('input', newValue);

    expect(wrapper).toEmit('update:showType', newValue);
  });

  test('Renders `availability-history-filters` with default props', () => {
    const wrapper = snapshotFactory();

    expect(wrapper).toMatchSnapshot();
  });

  test('Renders `availability-history-filters` with custom props', () => {
    const wrapper = snapshotFactory({
      propsData: {
        interval: QUICK_RANGES.yesterday,
        showType: AVAILABILITY_SHOW_TYPE.duration,
        minIntervalDate: 1000000000,
      },
    });

    expect(wrapper).toMatchSnapshot();
  });
});
