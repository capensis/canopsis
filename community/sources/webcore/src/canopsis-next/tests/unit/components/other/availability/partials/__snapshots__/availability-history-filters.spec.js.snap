// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`availability-history-filters Renders \`availability-history-filters\` with custom props 1`] = `
<div
  class="layout availability-history-filters gap-6 wrap"
  interval="[object Object]"
>
  <c-sampling-field-stub
    class="availability-history-filters__sampling"
    samplings="hour,day"
  />
  <availability-show-type-field-stub
    class="availability-history-filters__show-type"
    label="Show"
    name="availability"
    value="1"
  />
</div>
`;

exports[`availability-history-filters Renders \`availability-history-filters\` with default props 1`] = `
<div
  class="layout availability-history-filters gap-6 wrap"
>
  <c-sampling-field-stub
    class="availability-history-filters__sampling"
    samplings="hour,day"
  />
  <availability-show-type-field-stub
    class="availability-history-filters__show-type"
    label="Show"
    name="availability"
    value="0"
  />
</div>
`;
